From 4d4f67bfb0bbdc6dcaec444d0b276f15def6a0e5 Mon Sep 17 00:00:00 2001
From: Wolfgang Grandegger <wg@aries-embedded.de>
Date: Mon, 15 Dec 2025 11:13:52 +0100
Subject: [PATCH 6/6] stm32mp157x-msmp1evk: fix slew-rate setting for pins 4..7
 for the eMMC

The pinctrl setting for the pins 4..7 for the eMMC in the device-tree
have not been correct. Actually the following lines are missing:

  slew-rate = <1>;
  drive-push-pull;
  bias-pull-up;

With these settings, the double data rate with 1.8V and 50 MHz
(mmc-ddr-1_8v) does now work for the eMMC.

Signed-off-by: Wolfgang Grandegger <wg@aries-embedded.de>
---
 arch/arm/dts/stm32mp15-pinctrl.dtsi | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/dts/stm32mp15-pinctrl.dtsi b/arch/arm/dts/stm32mp15-pinctrl.dtsi
index 0184a22bde1..3543d46009b 100644
--- a/arch/arm/dts/stm32mp15-pinctrl.dtsi
+++ b/arch/arm/dts/stm32mp15-pinctrl.dtsi
@@ -1779,6 +1779,9 @@
 				 <STM32_PINMUX('A', 15, AF9)>, /* SDMMC2_D5 */
 				 <STM32_PINMUX('C', 6, AF10)>, /* SDMMC2_D6 */
 				 <STM32_PINMUX('D', 3, AF9)>; /* SDMMC2_D7 */
+			slew-rate = <1>;
+			drive-push-pull;
+			bias-pull-up;
 		};
 	};
 
-- 
2.34.1

